### ISSUE TITLE
Add support for block image target as data URI

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#3791]
There are various code paths for which a block image macro with a data URI target is not supported. One case is when both allow-uri-read and data-uri are set. Another one #3764. When the target of a block image macro is a data URI, that URI should be preserved, processed if necessary, and passed through to the output. (We should also consider whether this support should be extended to inline image macros).

Example:

```
image::data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAwAAAAMCAIAAADZF8uwAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH4wUbCB4vYn/PXAAAAB1pVFh0Q29tbWVudAAAAAAAQ3JlYXRlZCB3aXRoIEdJTVBkLmUHAAAAE0lEQVQY02NcxUAYMDGMKhqEigA8yQDCfg5F4QAAAABJRU5ErkJggg==[red square]
```

To support an SVG as a data URI, it may be necessary to specify the format explicitly.

```
image::data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyMDAgMjAwIj48cmVjdCB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCIgZmlsbD0iI2ZmMDAwMCI+PC9yZWN0Pjwvc3ZnPg==[red square,format=svg]
```

This requirement could be avoided if we look for and parse the data URI up front when looking for the image format.

For most images, the processor can assume the data is base64 encoded. However, for SVG images, the data can be passed through in raw XML. In this case, the data should be base64 encoded so it's properly escaped in the output.

When the format is SVG and opts=inline, the data should be read from the data URI, base64 decoded if necessary, and emitted as an embedded SVG.

**Comment by mojavelinux** [comment]
Note that currently, if the target of a block image macro is a data URI, the format is SVG, and opts=inline is set, the processor crashes.

**Comment by mojavelinux** [comment]
For what it's worth, this seems to already be supported. However, there may be situations where it's not fully supported. So the purpose of this issue would be to ensure it is supported in all cases.

This also works for inline images. However, in that case, it's necessary to escape the base64 data (e.g., `image:data:image/png;base64,pass:[...][alt]`) to avoid the processor from applying substitutions to it.

**Comment by ggrossetie** [comment]
> For what it's worth, this seems to already be supported. 

Indeed, basic `data:` URI targets already render. But here are two concrete cases where it's *not* fully supported.

#### When the user wants to inline SVG via a `data:` URI target

With `opts=inline`, an image whose target is a `data:image/svg+xml` URI is **not** decoded and embedded.
Asciidoctor falls back to `<img src="data:...">`, because the SVG detection only matches `format=svg` or a `.svg` target.

(As you noted, for an inline image the base64 payload must be wrapped in `pass:[...]` to keep the processor from applying substitutions to it.)

```bash
# Minimal SVG
cat > circle.svg <<'EOF'
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
  <circle cx="50" cy="50" r="40" fill="tomato"/>
</svg>
EOF

# Encode it as a base64 data URI
DATAURI="data:image/svg+xml;base64,$(base64 -i circle.svg | tr -d '\n')"

# AsciiDoc document using an inline image with opts=inline
cat > inline-svg-datauri.adoc <<EOF
= Inline SVG via data URI

Embedded diagram: image:${DATAURI}[Circle,opts=inline]
EOF

asciidoctor -s -a allow-uri-read -o - inline-svg-datauri.adoc
```

**Actual (2.0.26):**
```html
<p>Embedded diagram: <span class="image"><img src="data:image/svg+xml;base64,PHN2Zy..." alt="Circle"></span></p>
```
`opts=inline` is silently ignored.

**Expected:**
```html
<p>Embedded diagram: <span class="image"><svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 100 100">
  <circle cx="50" cy="50" r="40" fill="tomato"/>
</svg>
</span></p>
```
The data URI is decoded and the `<svg>` is embedded inline.

#### When the target is a `data` uri and `data-uri` (+ `allow-uri-read`) is set

In this case, `AbstractNode#image_uri` tries to re-fetch/re-encode it via OpenURI, which fails and logs a spurious warning.
The data URI is already an embedded image and should be passed through as-is.

```bash
PNG="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg=="
cat > raster-datauri.adoc <<EOF
= Raster data URI

image::${PNG}[Pixel]
EOF

asciidoctor -s -a data-uri -a allow-uri-read -o - raster-datauri.adoc
```

**Actual (2.0.26):**
```
asciidoctor: WARNING: could not retrieve image data from URI: data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAA...
```
(the `<img>` is still emitted, but a spurious warning is logged)

**Expected:**
No warning; the `data:` URI is passed through unchanged to `img src`.
